Fix ARM to install automate version 2 and extract inline bash into a script

gavindidrichsen opened the pull request:
Signed-off-by: Gavin Didrichsen <gavin.didrichsen@gmail.com>

gavindidrichsen committed:
Fix Azure ARM template process: (1) to install automate version 2 and
(2) to refactor the bash to a script file from an
inline concatenation of all statements

Signed-off-by: Gavin Didrichsen <gavin.didrichsen@gmail.com>

gavindidrichsen commented:
@skpaterson @russellseymour:  Can you guys review this PR?  Automate 2 is installing now via deployment process.  CC: @trevorghess, @davymcaleer.

gavindidrichsen committed:
Update chef-automate-install.sh

Fixed some indenting

skpaterson commented on scripts/chef-automate-install.sh:
```diff
@@ -0,0 +1,213 @@
+# set positional arguments in their proper place
+eval set -- "$PARAMS"
+
+# Set magic variables for current file & dir
```
Unless I'm mistaken, looks like these variables are defined but not used anywhere.  If that's the case could remove here.

gavindidrichsen replied:
I used PARAMS and the "magic variables" during the development and testing of the script, but can remove them now.

skpaterson commented on scripts/chef-automate-install.sh:
```diff
@@ -0,0 +1,213 @@
+}
+
+: '
+NOTE: Not sure that we need this anymore.
```
Would be good to get clarity on this either way and remove/add back to the main method accordingly.

skpaterson reviewed (changes requested):
Looks good Gavin 👍 approval only pending the couple of comments

gavindidrichsen committed:
Adjusted code in line with PR comments: (1) Removed unused magic
variables (but retained 2 because using them in log statements); (2) Removed the
unused PARAMS variable because the scipt only uses flagged parameters;
(3) Removed unnecessary software installation code

Signed-off-by: Gavin Didrichsen <gavin.didrichsen@gmail.com>

skpaterson reviewed (approved):
👍

gavindidrichsen committed:
Fix broken test for packages

Signed-off-by: Gavin Didrichsen <gavin.didrichsen@gmail.com>

russellseymour commented on azuredeploy.json:
```diff
@@ -265,8 +279,7 @@
     "chefAutoPublicIPAddressID": "[resourceId('Microsoft.Network/publicIPAddresses',variables('networkSettings').chefAutoPublicIPAddressName)]",
-    "prefix": "[uniqueString(resourceGroup().id)]",
-    "provider": "[toUpper('33194f91-eb5f-4110-827a-e95f640a9e46')]"
+    "prefix": "[uniqueString(resourceGroup().id)]"
```
Using just the `resourceGroup().id` to generate a unique string will not be enough. This is because if someone creates the same Resource Group name then their unique string will be the same as yours.

This is less of a problem now that Storage Accounts are not required as much as they used to be, however it is still good to increase the enrtopy a bit.

I have used this to create the unique string, which is then based on the subscription id, deployment name and the resource group. It can be further increased by adding a prefix, for example:

```
uniqueString(subscription().subscriptionId, resourceGroup().id, deployment().name, parameters('prefix'))
```
```

gavindidrichsen replied:
Hi Russell.  I've fixed this and committed.

russellseymour commented on azuredeploy.json:
```diff
@@ -245,8 +260,7 @@
-      "followerCustomData": "[base64(concat('#cloud-config\n', '\n', 'write_files:\n', '-   encoding: b64\n', '    content: ', base64(variables('scripts').chef_backend_install_script_follower) , '\n', '    path: /etc/chef-backend/chef-backend-install.sh\n', '    permissions: 0700\n'))]",
-      "automateCustomData": "[base64(concat('#cloud-config\n', '\n', 'write_files:\n', '-   encoding: b64\n', '    content: ', base64(variables('chef_automate_install_script_base')) , '\n', '    path: /etc/delivery/chef-automate-install.sh\n', '    permissions: 0700\n'))]"
+      "followerCustomData": "[base64(concat('#cloud-config\n', '\n', 'write_files:\n', '-   encoding: b64\n', '    content: ', base64(variables('scripts').chef_backend_install_script_follower) , '\n', '    path: /etc/chef-backend/chef-backend-install.sh\n', '    permissions: 0700\n'))]"
     },
```
You can use `customData`, which drives CloudInit, to set the FQDN of the machine
This then means you do not have to explicilty set it for Chef or Automate configuration.

For example if it is a front end machine you can use the PublicIP address FQDN (which can be retrieved in the ARM template) to set the FQDN in custom data.

gavindidrichsen replied:
I've added this as a task on the project board.  See [card](https://github.com/chef-partners/chef-automate-ha/projects/1#card-10831703)

russellseymour reviewed (changes requested):
I have added a couple of comments.

gavindidrichsen committed:
Increase the entropy on the "prefix" variable to ensure it is unique

Signed-off-by: Gavin Didrichsen <gavin.didrichsen@gmail.com>
